[DataGrid] Implement internationalization (#718)
mui-org/material-ui-x: docs/src/pages/components/data-grid/localization/localization.md

<p class="description">The Date Grid will support users from different locale, with formating, RTL, and localized strings.</p>

The default locale of Material-UI is English (United States). If you want to use other locales, follow the instructions below.

## Translation keys

You can use the `localeText` prop to pass in your own text and translations.

> You can find all the translation keys supported in [the source](https://github.com/mui-org/material-ui-x/blob/HEAD/packages/grid/_modules_/grid/constants/i18nConstants.ts) in the GitHub repository.

Reviewer: We should perhaps document them properly at some point.

Author: `localeText` prop is added to the api pages of both DataGrid and xGrid. If you are talking about the constant, I've exported an interface with the typing so that people can check it from the code.

Reviewer 2: > We should perhaps document them properly at some point.

@mbrookes We don't do it either for the core components. I have seen ag-Grid render the default value for the locale. But linking the page on GitHub with the correct version could be as good.

In the following example, the labels of the density selector are customized.

{{"demo": "pages/components/data-grid/localization/CustomLocaleTextGrid.js", "bg": "inline"}}

## 🚧 Locale text

The data grid will support dozens of locales with a simple import that includes all the translated messages.

> You will be able to follow [this guide](/guides/localization/#locale-text) to use them.

Reviewer: ```suggestion
You can follow [this guide](/guides/localization/#locale-text) to use them.
```

<p class="description">The Date Grid will support users from different locale, with formating, RTL, and localized strings.</p>

The default locale of Material-UI is English (United States). If you want to use other locales, follow the instructions below.

## Translation keys

You can use the `localeText` prop to pass in your own text and translations.

> You can find all the translation keys supported in [the source](https://github.com/mui-org/material-ui-x/blob/HEAD/packages/grid/_modules_/grid/constants/localeTextConstants.ts) in the GitHub repository.

Reviewer: I presume we plan to document this properly at some point?

Author: @oliviertassinari answered that in one of the comments before - [https://github.com/mui-org/material-ui-x/pull/718#discussion_r545002613](https://github.com/mui-org/material-ui-x/pull/718#discussion_r545002613)

Maybe we can look into documenting it after all the text is converted, although it may be an overkill since it is also in the code and a reference can work just as good.

<p class="description">The Date Grid will support users from different locale, with formating, RTL, and localized strings.</p>

## Locale text

The grid will support localization with a single prop that includes all the translated messages.
You will be able to follow [this guide](/guides/localization/#locale-text) to use them.

> You can use the `localeText` prop to pass in your own text and translations. The default translations are for en_EN.

Reviewer: ```suggestion
You can use the `localeText` prop to pass in your own text and translations. The default translations are for en_US.
```
